### ISSUE TITLE
Project won't compile after upgrade to 5.3.3

### ISSUE THREAD

**Issue by justasdev** [rsuite/rsuite-table#313]
### Versions

| package        | version |
| -------------- | ------- |
| `react`        | `17.0.2` |
| `rsuite-table` | `5.3.3` |

### What is the expected behavior?

Project should compile as before

### What is the current behavior?

Project won't compile

### What are the steps to reproduce?

Just update the rsuite-table version and build fails.

### Any additional comments? (optional)

Looks like this change causes the issue. The error is `An index signature parameter type must be either 'string' or 'number'.`
BTW thanks for fixing https://github.com/rsuite/rsuite-table/issues/304 so fast!

<img width="342" alt="image" src="https://user-images.githubusercontent.com/27926441/153624482-1ee04ec4-782d-4d89-b2f9-513d2cd6b48e.png">

Maybe this would fix the issue?
<img width="390" alt="image" src="https://user-images.githubusercontent.com/27926441/153633774-93a1cf9f-17e1-4990-981e-6173370d5254.png">

**Comment by simonguo** [comment]
hi @justasdev 
What version of typescript are you using? Try upgrading to the latest typescript, it should have been fixed.

**Comment by justasdev** [comment]
Our TS version is 4.3.5 and I see that support for symbols as indexes was added in 4.4 (August 2021). Haven't tried upgrading typescript version yet, it is not so trivial in a big project because of number of dependencies which some of them may not support the latest version of typescript. And I believe most of the projects would have exact same issue, because this feature is pretty new in typescript.
It looks as a breaking breaking change which should not go into patch version upgrade IMO, especially if there is a non-breaking alternative, I would go for it.
It's up to you though, we will either upgrade rsuite-table after upgrading typescript or fork this repo and fix it manually.

**Comment by simonguo** [comment]
![image](https://user-images.githubusercontent.com/1203827/155689086-3a68fe52-e658-4cb2-9653-1f8b8bb7f9e2.png)
Such a solution will still throw an error if strict mode is enabled in versions below TS 4.4.
```
 Type 'unique symbol' cannot be used as an index type.
```
My suggestion is to skip the check for node_modules . Modify skipLibCheck in `tsconfig.json`
```json
{
  "skipLibCheck": true
}
```

If you have a better solution to fix it you can submit a PR and I will release a new version for you.

**Comment by justasdev** [comment]
@simonguo  sorry for delayed response. `skipLibCheck` is not working too, because it works only with definition files.
We've just ignored this issue so far and not upgraded the lib.
However, I've investigated this issue and seems like the only solution is declaring symbol's type as string and casting to any. Probably it could be solved with further refactoring - not attaching data to row directly but making a map which maps rows to it's internal state (PARENT_KEY etc..), but it brings unnecessary complexity just to work around typescript [bug](https://github.com/Microsoft/TypeScript/issues/24587) . Here is a [PR](https://github.com/rsuite/rsuite-table/pull/329) for fixing it for those who uses this great table library with older typescript.
